azurerm/internal/services/mssql/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,503 @@
+		AutoPatchingSettings:                   expandArmSqlVirtualMachineAutoPatchingSettings(d.Get("auto_patching").([]interface{})),
+		KeyVaultCredentialSettings:             expandArmSqlVirtualMachineKeyVaultCredential(d.Get("key_vault_credential").([]interface{})),
+		ServerConfigurationsManagementSettings: expandArmSqlVirtualMachineServerConfigurationsManagement(d.Get("server_configuration").([]interface{})),
+	}
+
+	if sqlVirtualMachineGroupResourceID, ok := d.GetOk("sql_virtual_machine_group_id"); ok {
+		properties.SQLVirtualMachineGroupResourceID = utils.String(sqlVirtualMachineGroupResourceID.(string))
+	}
+
+	parameters := sqlvirtualmachine.SQLVirtualMachine{
+		Location:   utils.String(*respvm.Location),
+		Properties: &properties,
+		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroupName, name, "*")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q) ID", name, resourceGroupName)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmMsSqlVirtualMachineRead(d, meta)
+}
+
+func resourceArmMsSqlVirtualMachineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MSSQL.SQLVirtualMachinesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.MssqlVmID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroupName := id.ResourceGroup
+	name := id.Name
+
+	resp, err := client.Get(ctx, resourceGroupName, name, "*")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Sql Virtual Machine %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	if properties := resp.Properties; properties != nil {
+		d.Set("virtual_machine_id", properties.VirtualMachineResourceID)
+		d.Set("sql_license_type", string(properties.SQLServerLicenseType))
+		flattenedAutoPatching := flattenArmSqlVirtualMachineAutoPatching(properties.AutoPatchingSettings)
+		if err := d.Set("auto_patching", flattenedAutoPatching); err != nil {
```

Comment:
we can combine these lines
```suggestion
		if err := d.Set("auto_patching", flattenArmSqlVirtualMachineAutoPatching(properties.AutoPatchingSettings)); err != nil {
```